pavelib/quality.py

```diff
@@ -150,54 +150,74 @@ def _count_pylint_violations(report_file):
     return num_violations_report
 
 
-@task
-@needs('pavelib.prereqs.install_python_prereqs')
-@cmdopts([
-    ("system=", "s", "System to act on"),
-])
-def run_pep8(options):
+def _get_pep8_violations():
     """
-    Run pep8 on system code.
-    Fail the task if any violations are found.
+    Runs pep8. Returns a tuple of (number_of_violations, violations_string)
+    where violations_string is a string of all pep8 violations found, separated
+    by new lines.
     """
-    systems = getattr(options, 'system', ALL_SYSTEMS).split(',')
+    systems = ALL_SYSTEMS.split(',')
 
     report_dir = (Env.REPORT_DIR / 'pep8')
     report_dir.rmtree(ignore_errors=True)
     report_dir.makedirs_p()
-    
+
     # Make sure the metrics subdirectory exists
     Env.METRICS_DIR.makedirs_p()
 
     for system in systems:
         sh('pep8 {system} | tee {report_dir}/pep8.report -a'.format(system=system, report_dir=report_dir))
 
-    count = _count_pep8_violations(
+    count, violations_list = _pep8_violations(
         "{report_dir}/pep8.report".format(report_dir=report_dir)
     )
 
+    return (count, violations_list)
+
+
+def _pep8_violations(report_file):
+    """
+    Returns a tuple of (num_violations, violations_list) for all
+    pep8 violations in the given report_file.
+    """
+    with open(report_file) as f:
+        violations_list = f.readlines()
+    num_lines = len(violations_list)
+    return num_lines, violations_list
+
+
+@task
+@needs('pavelib.prereqs.install_python_prereqs')
+@cmdopts([
+    ("system=", "s", "System to act on"),
+])
+def run_pep8(_options):
+    """
+    Run pep8 on system code.
+    Fail the task if any violations are found.
+    """
+    (count, violations_list) = _get_pep8_violations()
+    violations_str = ''.join(violations_list)
+
     # Print number of violations to log
     violations_count_str = "Number of pep8 violations: {count}".format(count=count)
-    print(violations_count_str)
+    print violations_count_str
+    print violations_str
 
     # Also write the number of violations to a file
     with open(Env.METRICS_DIR / "pep8", "w") as f:
-        f.write(violations_count_str)
+        f.write(violations_count_str + '\n\n')
```

Comment:
Ah this happened because I solved a pylint error in an incorrect way. Fixed now.